sw/device/lib/dif/dif_alert_handler.c

```diff
@@ -4,9 +4,930 @@
+      return false;
+  }
+  mmio_region_write32(handler->params.base_addr, reg_offset,
+                      class->accumulator_threshold);
+
+  switch (class->alert_class) {
+    case kDifAlertHandlerClassA:
+      reg_offset = ALERT_HANDLER_CLASSA_TIMEOUT_CYC_REG_OFFSET;
+      break;
+    case kDifAlertHandlerClassB:
+      reg_offset = ALERT_HANDLER_CLASSB_TIMEOUT_CYC_REG_OFFSET;
+      break;
+    case kDifAlertHandlerClassC:
+      reg_offset = ALERT_HANDLER_CLASSC_TIMEOUT_CYC_REG_OFFSET;
+      break;
+    case kDifAlertHandlerClassD:
+      reg_offset = ALERT_HANDLER_CLASSD_TIMEOUT_CYC_REG_OFFSET;
+      break;
+    default:
+      return false;
+  }
+  mmio_region_write32(handler->params.base_addr, reg_offset,
+                      class->irq_deadline_cycles);
+
+  return true;
+}
+
+/**
+ * Configures phase durations of a particular class.
+ */
+DIF_WARN_UNUSED_RESULT
+static bool configure_phase_durations(
+    const dif_alert_handler_t *handler,
+    const dif_alert_handler_class_config_t *class) {
+  if (class->phase_durations == NULL && class->phase_durations_len != 0) {
+    return false;
+  }
+
+  for (int i = 0; i < class->phase_durations_len; ++i) {
+    // To save on writing a fairly ridiculous `if` chain, we use a lookup table
+    // that leverages the numeric values of enum constants.
+    static const ptrdiff_t kRegOffsets[4][4] = {
+            [kDifAlertHandlerClassA] =
+                {
+                    ALERT_HANDLER_CLASSA_PHASE0_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSA_PHASE1_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSA_PHASE2_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSA_PHASE3_CYC_REG_OFFSET,
+                },
+            [kDifAlertHandlerClassB] =
+                {
+                    ALERT_HANDLER_CLASSB_PHASE0_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSB_PHASE1_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSB_PHASE2_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSB_PHASE3_CYC_REG_OFFSET,
+                },
+            [kDifAlertHandlerClassC] =
+                {
+                    ALERT_HANDLER_CLASSC_PHASE0_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSC_PHASE1_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSC_PHASE2_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSC_PHASE3_CYC_REG_OFFSET,
+                },
+            [kDifAlertHandlerClassD] =
+                {
+                    ALERT_HANDLER_CLASSD_PHASE0_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSD_PHASE1_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSD_PHASE2_CYC_REG_OFFSET,
+                    ALERT_HANDLER_CLASSD_PHASE3_CYC_REG_OFFSET,
+                },
+    };
+
+    if (class->alert_class >= 4) {
+      return false;
+    }
+    dif_alert_handler_class_state_t phase = class->phase_durations[i].phase;
+    if (phase < kDifAlertHandlerClassStatePhase0 ||
+        phase > kDifAlertHandlerClassStatePhase3) {
```

Comment:
Please correct me if I am wrong, but I thought the consensus was we don't check the validity of an input if it is an input is an enum defined in the DIF header (I think you might have suggested it yourself)?

Reply:
I'm not sure if that's true, but this is checking that it's within a subset of the enum (since this enum contains other potential class states that aren't escalation phases).